### DISCUSSION TITLE
Prefix for 3rd party crates: bevyx

### DISCUSSION THREAD

**Discussion by Geams** [bevyengine/bevy#4256] [Ideas]
I would like to suggest a prefix for 3rd party crates analogous to deno.land:
bevy**x**_
Only official bevyengine crates should have the prefix bevy_.

**Comment by nicopap** [comment]
This is discussed in the plugin guidelines here https://github.com/bevyengine/bevy/blob/main/docs/plugins_guidelines.md#naming

As someone who maintains four crates with names starting with `bevy_`, I'd rather not have to go back and update them all :grimacing: Then there is the question of what happens to plugins made before the change in naming guideline. How do we get everyone to change the name of their crate?

I think things as they are right now are fine. Having the requirement to have "non-generic" names (on top of the bevy prefix) is enough. Currently only bevy has `bevy_singleword` crates. It's enough to distinguish non official crates. Honestly, I regret using the `bevy_` prefix in my crates, because then it makes for particularly long `use` statements. But also the prefix helps indicate the crate is specific to bevy and generally cannot be used in the rust ecosystem outside of bevy.
